### DISCUSSION TITLE
Custom Event Goal Tracking in Wordpress with Elementor

### DISCUSSION THREAD

**Discussion by foligo** [plausible/analytics#1752] [Anything Else]
Hey there,

I want to implement the custom event goal tracking ([https://plausible.io/docs/custom-event-goals](https://plausible.io/docs/custom-event-goals)) on my Wordpress website which runs Elementor. Things get kind of tricky as Elementor renders all custom link attributes ([https://elementor.com/help/custom-link-attributes/](https://elementor.com/help/custom-link-attributes/)) from wx|yz to wx="yz" but plausible says, "_Do watch the quotes! The data-analytics tag value should have both single and double quotes as shown above._"

Does anyone have some experience to share how she or he has managed this challenge?

Thanks a lot in advance and best wishes,

**Comment by Vigasaurus** [comment]
I can't say I've used Elementor, but from the docs it seems like you should be able to pass in `data-analytics|"Custom Event"` and it should get parsed as `<a data-analytics='"Custom Event"'>` (which is what you'd want). This is just speculation given my brief look at the Elementor docs, but it's worth a shot. 

Note: You might need to use single quotes instead in the example above - depending on how Elementor parses it and which quotes it already uses for wrapping the entire value.

**Reply by foligo** [reply]
Thx so much @Vigasaurus for your quick reply. 

When I paste data-analytics|"Custom Event" it get parsed to data-analytics|""Custom Event"" -> it looks like this when watch the code with inspector. But when I click in it, also in inspector, it changes to ' " Custom Event " '. And this works, so the event is sent to plausible.

But I'm not able to sent properties with it ex.: 
Insert: data-analytics|"kontakt-button", {"props":{"location":"header"}}"
Parsed: data-analytics='""kontakt-button"' props=""

Do you have any idea how to fix this issue?

Thx a lot and best wishes,

**Reply by Vigasaurus** [reply]
Unfortunately, I don't think you'd be able to do custom properties in this manner with Elementor, as to do custom properties you have to have a comma in there separating the goal name and the custom props, but this Elementor field will always parse the comma as the delimiter between two attributes, splitting it in the wrong place and breaking it. 

I don't know much about Elementor so I don't know if there's a different way to apply attributes, but this way almost certainly won't work for custom properties.

**Comment by GDXFred** [comment]
I`m also looking for a way to track events on a site running Elementor.
Any news regarding this?

**Comment by ThomasKoscheck** [comment]
Hi everyone 👋

Just wanted to check in on this — is there any update or workaround available?

With Elementor powering over 10 million active installations, I imagine there's significant demand for native event tracking with Plausible. I've been trying to find a clean solution for a while now without much luck.

Would love to help test if there's a beta or contribute to documentation if that's useful.

**Reply by Dan0sz** [reply]
Hi @ThomasKoscheck! Thanks for reaching out! Can you share an URL to an Elementor form as an example or at least examples of custom attributes you'd like to track? That way I can test if modifications need to be made to the plugin to add native support.

Looking forward to your reply!

**Reply by ThomasKoscheck** [reply]
Hi, 
you can check the button called "Partner werden" at the top right corner of this page: https://sec-auditor.com

I added the plausible data in Elementor as you can see in this screenshot
<img width="496" height="301" alt="image" src="https://github.com/user-attachments/assets/91420728-48f4-41cd-9f02-c70cb41b59c3" />

which renders this HTML
```html
<div class="elementor-element elementor-element-e4bc264 elementor-align-center elementor-widget__width-auto elementor-hidden-mobile plausible-event-name=registration elementor-widget elementor-widget-button" data-id="e4bc264" data-element_type="widget" data-e-type="widget" data-widget_type="button.default">
   <div class="elementor-widget-container">
      <div class="elementor-button-wrapper">
         <a class="elementor-button elementor-button-link elementor-size-sm" href="https://sec-auditor.com/partner/">
            <span class="elementor-button-content-wrapper">
               <span class="elementor-button-text">Partner werden</span>
	   </span>
	  </a>
	</div>
    </div>
</div>
```

**Reply by Dan0sz** [reply]
Hi @ThomasKoscheck,


I just tested this with Elementor Pro 4.1.1 and it does seem to work on my end!

Here's what I did:

- Added a button, by going to Add Element > Basic > Button.
  <img width="139" height="98" alt="afbeelding" src="https://github.com/user-attachments/assets/c31dbe87-aafc-4fcf-8372-95934d053b69" />
- Edit Button > Advanced
  <img width="314" height="120" alt="afbeelding" src="https://github.com/user-attachments/assets/bb8b6811-10e4-46d9-9b14-ab808c85ab85" />
- Added `plausible-event-name=test_param` to the Layout > CSS Classes option.
  <img width="277" height="43" alt="afbeelding" src="https://github.com/user-attachments/assets/cc3eb852-dfde-4592-8c3e-2021e42b8768" />

So, in your case, moving `plausible-event-name=registration` to the CSS Classes option should work!

If I missed anything, or if there's anything else to test, please let me know! I'm not super familiar with Elementor, so it's very possible I missed something! 😅
